### ISSUE TITLE
Real time command output

### ISSUE THREAD

**Issue by mimistina** [sshnet/SSH.NET#464]
Hi, I am using SSH.NET library in my code to run a few commands in my Linux Server but the text box shows me the output only after running them all, is there a way to stream the commands in real time?

**Comment by keithallenjackson** [comment]
Look into use the ShellStream class and subscribe to the DataReceived event. Be sure to Flush the stream after each command being written.

**Comment by mimistina** [comment]
Hi Keith thanks a lot! I tried it and it worked perfect, only having a few issues with buffer

**Comment by keithallenjackson** [comment]
No Problem. Currently, some Write commands use the underlying buffer, and some straight to the device using a Channel. Using the *Write(string)* and *WriteLine(string)* overloads will write straight to the channel without buffering. Be aware that one of my PLs (#458) will "fix" this by everything using the underlying buffer. Not finished, though, and will not be in the latest NuGet release.

The *DataReceived* event will give you data as soon as it is received from the channel without buffering. 

Please close the Issue if everything is working for you. Thanks!

**Comment by MikFa** [comment]
It might just be my own lack of knowledge on the subject, but on some devices it seems like the datarecieved is subscribed to, too late for it to catch the first messages after the createshell is called. 
This is what I do. 
```C#
shellStream = client.CreateShellStream("...", 1, 1, 1, 1, 1024 * 10);
shellStream.DataReceived += DataRecieved;
```

On some of the newer devices I deal with I recently ran into a problem where the incoming queue is populated too fast for the datarecieve to even be set by my program. And the handler is therefore null and won't trigger the event datareceived.

Don't know if this is normal behaviour, and the reason it wokred was just me "misusing" the library because of slow hardware/software without realising it.  

Didn't make a new issue because I feel this is very much related to the real time output topic and how it might trip up others. I also am not sure if it warents a new issue because I am unable to figure out if it is intended behaviour.
